infinite_tracing/test/infinite_tracing/connection_test.rb

```diff
@@ -38,6 +40,7 @@ def test_connection_initialized_after_connecting
           with_serial_lock do
             timeout_cap do
               with_config localhost_config do
+                NewRelic::Agent.agent.service.instance_variable_set(:@request_headers_map, {"NR-UtilizationMetadata"=>"test_metadata"})
 
```

Comment:
That's a good point. I added it everywhere we are checking the metadata contents to be consistent with other items expected in the metadata, but it could be pulled out into it's own test. I just was concerned about doing things differently because it could be confusing when looking at the tests now. But if you think pulling it into it's own test would be better, we can do that for sure.

Reply:
all-in-all, I think we went a little overboard with the consistency of what all we assert each test case.  In the hot moment we were creating all these tests, I overlooked that repetition of our test cases and now thinking some smart refactoring here can help us DRY up our tests and focus them a little better.

Reply:
Yeah, I totally agree about that. Since I'm going to spending some time refactoring our tests for infinite tracing, is this something we want to address in this PR, or wait and see how the general refactoring goes?

Reply:
Let's refactor now and this specifically.  I think it would be a good use of time next couple days you're here and a useful improvement to the test framework.

Reply:
Alright, i removed these from the existing tests, and added a test that is just checking the request_header_map data is included in the metadata.